hivemq4/base-image/docker-entrypoint.sh

```diff
@@ -43,6 +48,17 @@ fi
 echo "set bind address from container hostname to ${ADDR}"
 export HIVEMQ_BIND_ADDRESS=${ADDR}
 
+# Run entrypoint parts
+find "/docker-entrypoint.d/" -follow -type f -print | sort -V | while read -r f; do
+  if [ -x "$f" ]; then
+    echo >&3 "$0: running $f"
+    "$f"
+  else
+    echo >&3 "$0: sourcing $f"
+    . "$f"
+  fi
+done
+
```

Comment:
IMHO this should be moved right to the top of the file just before the license gets written

Reply:
Now we need to be careful with permissions, as both have different behaviours:

- `parent.sh`
```
#!/usr/bin/env bash

echo "-- start parent --"
echo "Setting Fizz"
FIZZ=BUZZ


echo "-- call child --"
./child.sh


echo "Foo $FOO"
echo "Fizz $FIZZ"

echo "-- source child --"

. child.sh

echo "Foo $FOO"
echo "Fizz $FIZZ"

echo "-- end parent --"
```

- `child.sh`
```
#!/usr/bin/env bash

echo "-- start child --"
echo "setting FOO"
FOO=BAR

echo "Foo $FOO"
echo "Fizz $FIZZ"
echo "-- end child --"
```

```sh
./parent.sh
-- start parent --
Setting Fizz
-- call child --
-- start child --
setting FOO
Foo BAR
Fizz
-- end child --
Foo
Fizz BUZZ
-- source child --
-- start child --
setting FOO
Foo BAR
Fizz BUZZ
-- end child --
Foo BAR
Fizz BUZZ
-- end parent --
```

So when sub-images are now adding files, depending on the permission of that file, the parent variables will be available or not. IMHO this can lead to confusing behaviour